internal/logic_handlers/organizationlogic/service_test.go

```diff
@@ -0,0 +1,680 @@
+				}
+				err = entry.Insert(context.Background(), db, boil.Infer())
+				require.NoError(t, err, "error inserting in the user db")
+			},
+			assertions: func(t *testing.T, paginated *model.PaginatedOrganizations, err error) {
+				require.NoError(t, err, "unexpected get organizations error")
+				require.NotNil(t, paginated, "unexpected nil organizations")
+				require.NotNil(t, paginated.Pagination, "unexpected nil pagination")
+				require.NotNil(t, paginated.Organizations, "unexpected nil organizations")
+				assert.Equal(t, 1, paginated.Pagination.RowCount, "unexpected total count")
+			},
+		},
+		{
+			name:            "fail-get-organizations",
+			getDependencies: getConcreteDependencies,
+			args: argsGetOrganizations{
+				ctx:    context.Background(),
+				filter: &model.OrganizationFilters{},
+			},
+			mutations: func(t *testing.T, db *sqlx.DB) {
+				testhelper.DropTable(t, db, mysqlmodel.TableNames.Organization)
+			},
+			assertions: func(t *testing.T, paginated *model.PaginatedOrganizations, err error) {
+				require.Error(t, err, "unexpected get organizations error")
+				require.Contains(t, err.Error(), "get organizations:")
+			},
+		},
+		{
+			name: "fail-mock-get-db",
+			getDependencies: func(t *testing.T) (*dependencies, func(ignoreErrors ...bool)) {
+				cleanup := func(ignoreErrors ...bool) {
+
+				}
+
+				mockTxProvider := persistencefakes.FakeTransactionProvider{}
+				mockTxProvider.DbReturns(nil, errors.New(mockDbReturnsErr))
+
+				return &dependencies{
+					Persistor:  &organizationlogicfakes.FakePersistor{},
+					TxProvider: &mockTxProvider,
+					Logger:     mockLogger,
+					Cleanup:    cleanup,
+				}, cleanup
+			},
+			args: argsGetOrganizations{
+				ctx:    context.Background(),
+				filter: &model.OrganizationFilters{},
+			},
+			mutations: func(t *testing.T, db *sqlx.DB) {},
+			assertions: func(t *testing.T, paginated *model.PaginatedOrganizations, err error) {
+				require.Error(t, err, "unexpected get organizations error")
+				require.Contains(t, err.Error(), "get db:")
+			},
+		},
+	}
+}
+
+func TestService_GetOrganizations(t *testing.T) {
+	for _, tt := range getGetOrganizationTestCases() {
+		t.Run(tt.name, func(t *testing.T) {
+			_dependencies, cleanup := tt.getDependencies(t)
+			defer cleanup()
+
+			svc, err := New(&Config{
+				TxProvider: _dependencies.TxProvider,
+				Logger:     _dependencies.Logger,
+				Persistor:  _dependencies.Persistor,
+			})
+			require.NoError(t, err, "unexpected new error")
+
+			tt.mutations(t, _dependencies.Db)
+
+			paginatedOrganizations, err := svc.ListOrganizations(tt.args.ctx, tt.args.filter)
+			tt.assertions(t, paginatedOrganizations, err)
+		})
+	}
+}
+
+type testCaseUpdateOrganization struct {
+	name            string
+	getDependencies func(t *testing.T) (*dependencies, func(ignoreErrors ...bool))
+	ctx             context.Context
+	mutations       func(t *testing.T, db *sqlx.DB) (toUpdatedOrganization *model.UpdateOrganization)
```

Comment:
as convention, let's not used named returns so:

func(t *testing.T, db *sqlx.DB) (toUpdatedOrganization *model.UpdateOrganization)

should be

func(t *testing.T, db *sqlx.DB) *model.UpdateOrganization